src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/ExtensionDataTests.cs

```diff
@@ -466,6 +480,17 @@ public static void ExtensionProperty_IgnoresCustomSerializerWithOptions(Type ove
                 .Invoke(null, null);
         }
 
+        [Fact]
+        public static void ExtensionProperty_IgnoresCustomSerializerWithOptions_JsonObject()
+        {
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new JsonObjectOverflowConverter());
+
+            // A custom converter for JsonObject is not allowed on an extension property.
+            Assert.Throws<InvalidOperationException>(() =>
```

Comment:
Validate exception message?

Reply:
Do we generally validate for exception messages?

Reply:
Sometimes to ensure the correct message is includes. We only validate type name and other things that won't be localized (if\when we decide to localize exception strings in core).

Reply:
The exception didn't actually have its own message since it is a super edge case. However, I added the message and modified the test to validate it.